File: frontend/projects/api/src/lib/models/check-in-dto.ts
/**
 * A daily check-in. `date` is the server-derived America/Toronto local day
 * (`yyyy-MM-dd`, ADR-004) — clients never send it for today-operations.
 */
export interface CheckInDto {
  readonly id: string;
  readonly date: string;
  readonly moodRating: number;
  readonly spiritualRating: number;
  readonly note: string | null;
}

File: backend/src/Shalom.Application/Contracts/CheckInDto.cs
namespace Shalom.Application.Contracts;

/// <summary>A daily check-in. <paramref name="Date"/> is the server-derived America/Toronto local day (ADR-004).</summary>
public record CheckInDto(Guid Id, DateOnly Date, int MoodRating, int SpiritualRating, string? Note);
